File: 462-minimum-moves-to-equal-array-elements-ii/462-minimum-moves-to-equal-array-elements-ii.java
class Solution {
    public int minMoves2(int[] nums) {
        Arrays.sort(nums);
        
        int i = 0, j = nums.length - 1;
        int ans = 0;
        
        while(i < j) {
            ans += nums[j] - nums[i];
            i++;
            j--;
        }
        
        return ans;
    }
}

File: 462-minimum-moves-to-equal-array-elements-ii/462-minimum-moves-to-equal-array-elements-ii.cpp
class Solution {
public:
    int minMoves2(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        
        int median = 0;
        int n = nums.size();
        
        if(n%2 == 0)
            median = (nums[n/2] + nums[(n-1)/2])/2;
        else
            median = nums[n/2];
        
        int ans = 0;
        
        for(int i = 0; i < n; i++)
            ans += abs(nums[i] - median);
        
        return ans;
    }
};